cpp/src/arrow/compare.cc

```diff
@@ -154,10 +155,10 @@ class RangeEqualsVisitor {
         return false;
       }
       if (is_null) continue;
-      const int32_t begin_offset = left.value_offset(i);
-      const int32_t end_offset = left.value_offset(i + 1);
-      const int32_t right_begin_offset = right.value_offset(o_i);
-      const int32_t right_end_offset = right.value_offset(o_i + 1);
+      const auto begin_offset = left.value_offset(i);
```

Comment:
it is possible to templatize this instead of using auto?  I'm 50/50 on whether this would actually be a good change?

Reply:
Hmm, I don't really see the point...